### PR TITLE
[DOC] Add docs for ROOT::EnableThreadSafety

### PR DESCRIPTION
**Pull request by bluehood**

### CODE REVIEW

File: core/base/src/TROOT.cxx [hunk 1]
```diff
@@ -526,8 +526,23 @@ namespace Internal {
       return macroPath;
    }
 
+   // clang-format off
    ////////////////////////////////////////////////////////////////////////////////
    /// Enables the global mutex to make ROOT thread safe/aware.
+   ///
+   /// The following becomes safe:
+   /// - concurrent construction and destruction of TObjects, including the ones registered in ROOT's global lists (e.g. gListOfCleanups, gListOfFunctions and gListOfFiles)
```
Comment (pcanal) [inline core/base/src/TROOT.cxx]: gListOfCleanups is not the actually name gROOT->GetListOfCleanups(), etc ...

File: core/base/src/TROOT.cxx [hunk 2]
```diff
@@ -526,8 +526,23 @@ namespace Internal {
...
+   /// - concurrent usage of _different_ ROOT objects from different threads, including ones with global state (e.g. TFile, TTree, TChain) with the exception of graphics classes (e.g. TCanvas)
+   /// - concurrent calls to ROOT's type system classes, e.g. TClass and TEnum
+   /// - concurrent calls to the interpreter through gInterpreter
+   /// - concurrent loading of ROOT plug-ins
+   ///
+   /// In addition, gDirectory becomes a thread-local variable which defaults to a TDirectory singleton.
```
Comment (pcanal) [inline core/base/src/TROOT.cxx]: gDirectory and gFile

Comment (pcanal) [reply core/base/src/TROOT.cxx]: and gPad

File: core/base/src/TROOT.cxx (hunk 1 above)
Comment (bluehood) [reply core/base/src/TROOT.cxx]: fixed

File: core/base/src/TROOT.cxx (hunk 2 above)
Comment (bluehood) [reply core/base/src/TROOT.cxx]: fixed

File: core/base/src/TROOT.cxx [hunk 3]
```diff
@@ -526,8 +526,23 @@ namespace Internal {
...
    /// Enables the global mutex to make ROOT thread safe/aware.
+   ///
+   /// The following becomes safe:
+   /// - concurrent construction and destruction of TObjects, including the ones registered in ROOT's global lists (e.g. gROOT->GetListOfCleanups(), gROOT->GetListOfFiles())
+   /// - concurrent usage of _different_ ROOT objects from different threads, including ones with global state (e.g. TFile, TTree, TChain) with the exception of graphics classes (e.g. TCanvas)
+   /// - concurrent calls to ROOT's type system classes, e.g. TClass and TEnum
+   /// - concurrent calls to the interpreter through gInterpreter
+   /// - concurrent loading of ROOT plug-ins
+   ///
+   /// In addition, gDirectory, gFile and gPad become a thread-local variable which defaults to a TDirectory singleton.
```
Comment (pcanal) [inline core/base/src/TROOT.cxx]: > defaults to a TDirectory singleton.

it is not an 'unamed' singleton.  It defaults to gROOT.

Comment (bluehood) [reply core/base/src/TROOT.cxx]: I'm confused, `gROOT` is not a `TDirectory`..?

Comment (pcanal) [reply core/base/src/TROOT.cxx]: It is :) so your sentence is correct and accurate :).  But gROOT is also 'well known' so why not mention 'which'  TDirectory singleton is being used :)

Comment (bluehood) [reply core/base/src/TROOT.cxx]: oh damn `TROOT` inherits from `TDirectory`. my bad.

Comment (bluehood) [reply core/base/src/TROOT.cxx]: wait...`gDirectory`, `gFile` and `gPad` all default to `gROOT`?

Comment (pcanal) [reply core/base/src/TROOT.cxx]: no, just gDirectory ....

checking what gFile and gPad default to ...

Comment (pcanal) [reply core/base/src/TROOT.cxx]: the other default to nullptr (if I read the code correctly).

Comment (bluehood) [reply core/base/src/TROOT.cxx]: ok thanks, fixed

**Pull request merged**